### PR TITLE
Bpf 2882v2

### PR DESCRIPTION
**Pull request by giretyu**
### What problem is this solving?

Create a new form that specifically handle Add Address Manually, 

RenderAddAddressManual render the manual address form, and we clear it every time 

https://bpf2992--thefoschini.myvtex.com/checkout/#/shipping

Manual form is rendered as a different section

Screenshot 2024-08-07 at 08 47 02

Form 

Screenshot 2024-08-08 at 13 16 39

Getting the native validation

Screenshot 2024-08-08 at 13 17 25

Phone Number validation
Screenshot 2024-08-08 at 13 32 19

Order form 

https://bpf2992--thefoschini.myvtex.com/api/checkout/pub/orderForm/a2160612d7a8460ab8735ca478cd6767

Screenshot 2024-08-08 at 13 21 34

Master Data

Screenshot 2024-08-08 at 13 25 44

#### How to test it?

[Workspace]()

### Screenshots/Video example usage:
- Desktop
- Tablet
- Movil

### CODE REVIEW

File: checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
+import FormField from './Elements/FormField';
+import { getBestRecipient } from './utils';
+
+const AddressForm = () => {
```
Comment (giretyu) [inline checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js:4]: change name to -> AddAddressManualForm

File: checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+  const formFields = fields.map((field) => FormField(field)).join('');
+
+  return `
+  <div>
+      <div class="bash--heading">
+        <h3>Add a new delivery address</h3>
+        <a href='#' data-view='select-address' id='back-button-select-address'>&lt; Back</a>
+      </div>
+    <form id="bash--address-form" method="post">
```
Comment (giretyu) [inline checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js:189]: change id of the form

File: checkout-ui-custom/src/controller/DeliverController.js [hunk 3]
```diff
@@ -63,10 +64,18 @@ const DeliverController = (() => {
     preparePhoneField(`#${EDIT_FORM_RECEIVER_PHONE_ID}`);
```
Comment (giretyu) [inline checkout-ui-custom/src/controller/DeliverController.js:64]: note the validation

File: checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js (hunk 2 above)
Comment (giretyu) [reply checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js:189]: check editForm submit logic

File: checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    {
+      name: 'receiverPhone',
+      label: `Recipient's mobile number`,
+      required: true,
+      type: 'tel',
+      helperText: 'We send shipping updates to this number.',
+      minlength: 9,
+      error: 'Please enter a valid phone number',
+      containerClasses: 'custom-field-complement',
+      idOverride: ADD_FORM_RECEIVER_PHONE_ID,
```
Comment (giretyu) [inline checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js:171]: rename to something better

File: checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+};
+
+export const submitAddAddressManualForm = async (event) => {
+  event.preventDefault();
+
+  // PULL ALL FORM FIELDS
+  const form = document.getElementById('bash--add-address-manual-form');
+  const formData = new FormData(form);
+
+  const payload = {};
```
Comment (giretyu) [inline checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js:191]: street it is a conct of number + street

File: checkout-ui-custom/src/controller/DeliverController.js [hunk 6]
```diff
@@ -63,10 +64,18 @@ const DeliverController = (() => {
...
   };
 
+  const RenderAddAddressManual = async () => {
+    document.querySelector('#manual-address-section').innerHTML = AddAddressManual();
```
Comment (shaheen-karodia) [inline checkout-ui-custom/src/controller/DeliverController.js:68]: Have you tested the validation on your form field: you need to prepare the Field

File: checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js (hunk 4 above)
Comment (shaheen-karodia) [reply checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js:171]: ADD_ADDRESS_FORM_MANUAL_RECIEVER_PHONE_ID

File: checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+      value: '',
+    },
+    {
+      name: 'lng',
+      required: false,
+      type: 'hidden',
+      value: '',
+    },
+    {
+      name: 'number',
```
Comment (shaheen-karodia) [inline checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js:54]: image

no seperation between street and number in designs 

https://www.figma.com/design/M9vAdnEoGENzSD3OkaE7BB/%5BWEB%5D-Cart-%26-Checkout?node-id=901-244&t=VhMljcQxaMHMRNrz-0

**Review by shaheen-karodia — CHANGES_REQUESTED** [review]

**Comment by shaheen-karodia** [comment]
@Gift-retyu 
In the PR  I need to see what happens with form validations
Want to see what happens on the orderform
What happens in master. data

File: checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+        { value: 'GP', label: 'Gauteng' },
+        { value: 'KZN', label: 'KwaZulu-Natal' },
+        { value: 'LP', label: 'Limpopo' },
+        { value: 'MP', label: 'Mpumalanga' },
+        { value: 'NC', label: 'Northern Cape' },
+        { value: 'NW', label: 'North West' },
+        { value: 'WC', label: 'Western Cape' },
+      ],
+    },
+    {
```
Comment (shaheen-karodia) [inline checkout-ui-custom/src/partials/Deliver/AddAddressManualForm.js:130]: if you look at the design: no need for a country display, the next field already captures 'ZAF'
image

**Review by shaheen-karodia — APPROVED** [review]
LGTM: you can change that one comment, I dont need to rereview

**Review by SegopotsoMakhutja — APPROVED** [review]
🌶️👌

**Pull request merged**